### PR TITLE
Fix: constraints selector dropdown appears behind lower tiles

### PR DESCRIPTION
**Pull request by toddburnside**

### CODE REVIEW

File: common/src/main/scala/explore/model/reusability.scala [hunk 1]
```diff
@@ -59,4 +60,5 @@ object reusability {
   implicit def modelUndoStacksReuse[F[_]]: Reusability[ModelUndoStacks[F]]           = Reusability.derive
   // Move to lucuma-ui
   implicit val semesterReuse: Reusability[Semester]                                  = Reusability.derive
+  implicit val cssReuse: Reusability[Css]                                            = Reusability.by(_.htmlClass)
```
Comment (toddburnside) [inline common/src/main/scala/explore/model/reusability.scala:63]: I was surprised I couldn't find a reuseability for Css anywhere. I can move this and the above to lucuma-ui and cut a release of it. (Unless I missed an instance somewhere)

Comment (cquiroz) [reply common/src/main/scala/explore/model/reusability.scala:63]: Maybe it should be in scalajs-react-common

File: explore/src/main/scala/explore/tabs/ObsTabContents.scala [hunk 2]
```diff
@@ -413,14 +413,19 @@ object ObsTabContents {
                   props.searching,
                   state.zoom(State.options)
                 ),
+                // The ExploreStyles.ConstraintsTile css adds a z-index to the constraints tile react-grid wrapper
+                // so that the constraints selector dropdown always appears in front of any other tiles. If more
+                // than one tile ends up having dropdowns in the tile header, we'll need something more complex such
+                // as changing the css classes on the various tiles when the dropdown is clicked to control z-index.
```
Comment (cquiroz) [inline explore/src/main/scala/explore/tabs/ObsTabContents.scala:419]: Neat

File: common/src/main/scala/explore/model/reusability.scala (hunk 1 above)
Comment (toddburnside) [reply common/src/main/scala/explore/model/reusability.scala:63]: Actually, I did find an instance, but only in seqexec

**Review by cquiroz — APPROVED** [review]
Great investigative work to solve this one, thanks

File: common/src/main/scala/explore/model/reusability.scala (hunk 1 above)
Comment (toddburnside) [reply common/src/main/scala/explore/model/reusability.scala:63]: > Maybe it should be in scalajs-react-common

Oh, good point.

**Review by rpiaggio — APPROVED** [review]
LGTM thanks for tackling this tricky issue

**Pull request merged**